### PR TITLE
spec for import cluster

### PR DESCRIPTION
**Pull request by gnehapk**

### CODE REVIEW

**Comment by gnehapk** [comment]
Against #56

**Comment by gnehapk** [comment]
@brainfunked @kamleshverma1 @nthomas-redhat @anivargi @shtripat Please review it.

File: specs/import-cluster-view.adoc [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+* On success, a successful message will be displayed to user.
+  (https://redhat.invisionapp.com/share/6T900V2ZX#/screens/198651334)
+
+== Proposed change
+
+Front-end related changes:
+
+	* Create views for Landing page (First Time Experience), Import cluster page and page where successful message will be displayed.
+	* Implement client side filer and sorting functionality using angular's filter and orderby components.
```
Comment (gnehapk) [inline specs/import-cluster-view.adoc]: @julienlim What are the options for sorting and filtering on import cluster view - https://redhat.invisionapp.com/share/6T900V2ZX#/screens/198042641

**Comment by mkudlej** [comment]
Please add https://github.com/Tendrl/usmqe-tests/issues/20 as reference for testing.

**Review by anivargi — APPROVED** [review]
Looks good.

File: specs/import-cluster-view.adoc (hunk 1 above)
Comment (julienlim) [reply specs/import-cluster-view.adoc]: @gnehapk 

The following are the options for sort and filter for the Import Cluster view.

Filter (by): name, IP address, role (if available for Ceph)
Sort (by): name, IP address, role (if available for Ceph)

Comment (shtripat) [inline specs/import-cluster-view.adoc]: s/filer/filter/g

**Review by shtripat — COMMENTED** [review]
LGTM. One minor spelling mistake.

File: specs/import-cluster-view.adoc [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+None
+
+=== Data model impact:
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+Not yet decided.
```
Comment (gnehapk) [inline specs/import-cluster-view.adoc]: @anivargi Please provide the APIs and their response which will be used in Import cluster functionality.

File: specs/import-cluster-view.adoc [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+== Use Cases
+
+This addresses the use case of importing existing clusters(ceph. gluster etc.) into tendrl for management.
+
+The workflow is as follows:
+
+* "Import Cluster" can be triggered from 
+
+	- Landing page (First Time Experience) if no cluster is present in tendrl
```
Comment (cybernth) [inline specs/import-cluster-view.adoc]: Does the landing page have this option if atleast one cluster is present? I assume it lands into the dashboard in that case

File: specs/import-cluster-view.adoc [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+	  (https://redhat.invisionapp.com/share/6T900V2ZX#/screens/198042644)
+
+	- Cluster List View if minimum one cluster is present in tendrl.
+	  (https://redhat.invisionapp.com/share/BR8JDCGSQ#/screens/185937524)
+
+* User clicks on "Import Cluster". 
+
+* It will take to host list page where user will select the hosts he wants to import.
```
Comment (cybernth) [inline specs/import-cluster-view.adoc]: User selects a single host participating in a cluster. Upon clicking 'Next" the API gets a list of hosts participating in the cluster. I think we need to reword

File: specs/import-cluster-view.adoc [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+  (https://redhat.invisionapp.com/share/6T900V2ZX#/screens/198651334)
+
+== Proposed change
+
+Front-end related changes:
+
+	* Create views for Landing page (First Time Experience), Import cluster page and page where successful message will be displayed.
+	* Implement client side filer and sorting functionality using angular's filter and orderby components.
+	* Send API call for host list whenevr users click "Refresh" button.
+	* Show total count of hosts which can be imported.
```
Comment (cybernth) [inline specs/import-cluster-view.adoc]: What does this mean?

**Review by cybernth — CHANGES_REQUESTED** [review]

File: specs/import-cluster-view.adoc (hunk 3 above)
Comment (gnehapk) [reply specs/import-cluster-view.adoc]: @nthomas-redhat When the application gets loaded, the 1st call we make is for get cluster list call. Based on its reponse, the app decides whether it should go to landing page or cluster list view.

**Comment by gnehapk** [comment]
@nthomas-redhat @brainfunked @anivargi @shtripat @kamleshverma1 I have updated the PR. Please review.

**Comment by kamleshverma1** [comment]
@julienlim As we(front-end guys) have discussed with Nishanth and mrugesh, we are not going to consider bootstrap node concept while importing cluster. All clusters will be auto detected at the time of import. and in case of multiple cluster , we will have some "select drop-down" from where user can select appropriate cluster which they want to be import. or You can suggest if there is any other alternative instead of using "select drop-down" in case of multiple cluster .

here i attached screen-shot for import cluster
screenshot from 2017-01-13 11 58 38

**Comment by kamleshverma1** [comment]
@julienlim i want some clarification on primary navigation visibility while importing cluster .
i assumed that basically there is two case for import cluster 

1) If there is at least one cluster in Tendrl - primary navigation should be visible so that user can navigate into any other pages because we have at least one cluster in this case.

2) If there is no any cluster in Tendrl - primary navigation should be hidden because there is no point to show navigation as we don't have anything (clusters, pools, file shares etc ) in this case.

please correct me if i am wrong

**Comment by kamleshverma1** [comment]
@anivargi I have gone through response format which you sent for nodelist and clusterlist .
and i observed that "stats" coming as string . 

"stats":"{'alert_cnt': 10, 'storage': {'total': 42949672960, 'used': 21474836480, 'updated_at': '2016-12-19T23:12:51.278616', 'percent_used': 50}, 'cpu': {'updated_at': '2016-12-19T23:12:51.278516', 'percent_used': 4.5}, 'memory': {'total': 4294967296, 'used': 2147483648, 'updated_at': '2016-12-19T23:12:51.278593', 'percent_used': 50}}"

Ideally it should be object instead of string in order to access the value of it . otherwise we need to do some extra logic unnecessary to get the value of it in UI.

**Pull request closed without merging**